### ISSUE TITLE
Terraform issue

### ISSUE THREAD

**Issue by mattjames1978** [jamesrcounts/terraform-getting-started-azure#6]
![5850C548-B719-416E-86B1-35C909118A5B](https://user-images.githubusercontent.com/97545090/155017960-bbabeb48-02ed-4cee-afb8-ca58798ad11e.jpeg)

Hi James, I wonder if you can help. I have had a go at deploying your code as is but running into an issue with the main.tf. 

Error: invalid index
Data.azurerm_resource_group.rg.tags is a map of string with element

The given key does not identify an element in this collection value.

Do I need to specify any particular tags on the resource group? 

Great article though.
Thanks
Matt

**Comment by jamesrcounts** [comment]
You can replace the reference to the resource group with any unique identifier you like for instance id.  For example:

instance_id = “matt_2022”

Or you can add a tag called instance_id on the resource group with the same value and leave the terraform code as is.

Personally I use a random_pet resource to generate the instance_id, but I use a different project to create mini “landing zones” and tag them which is why we don’t see it setup in the code you’re looking at.

TL;DR its just a string meant to help resources have a unique name.

Hope this helps!

From: mattjames1978 ***@***.***>
Reply-To: jamesrcounts/terraform-getting-started-azure ***@***.***>
Date: Monday, February 21, 2022 at 11:58 AM
To: jamesrcounts/terraform-getting-started-azure ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [jamesrcounts/terraform-getting-started-azure] Terraform issue (Issue #6)


[Image removed by sender. 5850C548-B719-416E-86B1-35C909118A5B]<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F97545090%2F155017960-bbabeb48-02ed-4cee-afb8-ca58798ad11e.jpeg&data=04%7C01%7C%7Cfe39425bf86f4e6dffe008d9f5747773%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637810702831196778%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=QEzTCO2rTvzXssHMBurBIqlQfCEqXBMWvCBESQvxeY0%3D&reserved=0>

Hi James, I wonder if you can help. I have had a go at deploying your code as is but running into an issue with the main.tf.

Error: invalid index
Data.azurerm_resource_group.rg.tags is a map of string with element

The given key does not identify an element in this collection value.

Do I need to specify any particular tags on the resource group?

Great article though.
Thanks
Matt

**Comment by mattjames1978** [comment]
Hi Jim, that is great thanks. All sorted now and thanks for the pointer.

I will have a play around with the random resource but hardcoding the instance ID in the RG tags got the ADO plan running.

Really enjoying your articles and it’s really helping me on my Devops learning journey! Keep up the good work 😊


***@***.***


____________________________________________
Matt James

Azure Cloud Architect | Capgemini UK

Capgemini UK | Woking
Forge End, Woking, GU21 6DB
Mob.: + 44 7467892486
www.capgemini.com<http://www.capgemini.com/>

Email ***@***.******@***.***>
***@***.***
[A picture containing text  Description automatically generated]<https://www.capgemini.com/resources/get-the-future-you-want>
____________________________________________
Connect with Capgemini:
***@***.***<https://twitter.com/Capgemini> ***@***.*** <https://www.facebook.com/Capgemini/>  ***@***.*** <https://www.linkedin.com/company/capgemini/mycompany/verification/>  ***@***.*** <https://www.youtube.com/profile?user=capgeminimedia>  ***@***.*** <https://www.slideshare.net/capgemini>  ***@***.*** <https://soundcloud.com/capgemini>  ***@***.*** <https://www.instagram.com/capgemini/>  ***@***.*** <https://www.glassdoor.fr/Overview/Working-at-Capgemini-EI_IE3803.11,20.htm?countryRedirect=true>

Please consider the environment and do not print this email unless absolutely necessary.
Capgemini encourages environmental awareness.
